plugins/turing/runner/experiment_runner.go

```diff
@@ -0,0 +1,151 @@
+package runner
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gojek/turing/engines/experiment/log"
+	"github.com/gojek/turing/engines/experiment/pkg/request"
+	inproc "github.com/gojek/turing/engines/experiment/plugin/inproc/runner"
+	"github.com/gojek/turing/engines/experiment/runner"
+	"github.com/pkg/errors"
+
+	xpclient "github.com/gojek/xp/clients/treatment"
+	"github.com/gojek/xp/plugins/turing/config"
+)
+
+// init ensures this runner is registered when the package is imported.
+func init() {
+	err := inproc.Register("xp", NewExperimentRunner)
+	if err != nil {
+		log.Panicf("failed to register xp experiment runner: %v", err)
+	}
+}
+
+// experimentRunner implements runner.ExperimentRunner
+type experimentRunner struct {
+	httpClient *xpclient.ClientWithResponses
+	projectID  int
+	passkey    string
+	parameters []config.RequestParameter
+}
+
+func (er *experimentRunner) GetTreatmentForRequest(
+	reqHeader http.Header,
+	body []byte,
+	options runner.GetTreatmentOptions,
+) (*runner.Treatment, error) {
+	logger := log.With("turing_req_id", options.TuringRequestID)
+
+	// Get the request parameters for the current request
+	requestParams := er.getRequestParams(logger, reqHeader, body)
+
+	// Create Request Payload
+	reqPayload, err := json.Marshal(requestParams)
```

Comment:
Marshal `map[string]string` instead of `map[string]interface{}` since Experiment Engine plugin should not know about Segmenter type conversions.